tests/system/action/speaker/test_create_point_of_order.py

```diff
@@ -189,8 +226,163 @@ def test_create_poo_not_activated_in_meeting(self) -> None:
             ],
         )
         self.assert_status_code(response, 400)
-        self.assert_model_not_exists("speaker/1")
         self.assertIn(
-            "Point of order speakers are not enabled for this meeting.",
+            "data must contain [\\'list_of_speakers_id\\', \\'user_id\\'] properties",
             str(response.data),
         )
+
+    def test_create_after_existing_poo(self) -> None:
+        self.create_model(
+            "meeting/7844",
+            {
+                "name": "name_asdewqasd",
+                "list_of_speakers_enable_point_of_order_speakers": True,
+                "list_of_speakers_present_users_only": False,
+            },
+        )
+        self.create_model("user/7", {"username": "talking with poo"})
+        self.create_model("user/8", {"username": "waiting with poo"})
+        self.update_model(
+            "user/1",
+            {
+                "speaker_$7844_ids": [3],
+                "speaker_$_ids": ["7844"],
+            },
+        )
+        self.create_model(
+            "speaker/1",
+            {
+                "user_id": 7,
+                "list_of_speakers_id": 23,
+                "point_of_order": True,
+                "begin_time": 100000,
+                "weight": 1,
+            },
+        )
+        self.create_model(
+            "speaker/2",
+            {
+                "user_id": 8,
+                "list_of_speakers_id": 23,
+                "weight": 2,
+                "point_of_order": True,
+            },
+        )
+        self.create_model(
+            "speaker/3", {"user_id": 1, "list_of_speakers_id": 23, "weight": 3}
+        )
+        self.create_model(
+            "list_of_speakers/23", {"speaker_ids": [1, 2, 3], "meeting_id": 7844}
+        )
+        response = self.client.post(
+            "/",
+            json=[
+                {
+                    "action": "speaker.create",
+                    "data": [
+                        {
+                            "user_id": 1,
+                            "list_of_speakers_id": 23,
+                            "point_of_order": True,
+                        }
+                    ],
+                }
+            ],
+        )
+        self.assert_status_code(response, 200)
+        self.assert_model_exists(
+            "speaker/2",
+            {
+                "user_id": 8,
+                "weight": 2,
```

Comment:
This test is wrong: speaker 2 and 4 have the same weight. The weight must be unique withing all waiting speakers.

In this test I would expect the weight to be as this:
- speaker/1: 1 (unaffected, since not waiting)
- speaker/2: 2 (not changed)
- speaker/3: 4 (must be higher that every waiting poo speaker)
- speaker/4: 3 (max weight of poos + 1)

Please have a look at https://github.com/OpenSlides/openslides-backend-coordination/wiki/List-of-speakers: I've added a more detailed description of the weight calculation.

Reply:
We should have a talk. See my alternative example(s) in the wiki, link above. If there is a POOS in the list, I understood, a new one should be after that.

Reply:
Done